deleting-compaction-strategy/src/main/java/com/protectwise/cassandra/db/compaction/KafkaBackupSinkForDeletingCompaction.java

```diff
@@ -0,0 +1,263 @@
+		// we don't need to bother resorting the data.
+		columnFamily = ArrayBackedSortedColumns.factory.create(cfs.keyspace.getName(), cfs.getColumnFamilyName());
+
+		this.cassandraPurgedKafkaTopic = cassandraPurgedKafkaTopic;
+
+		this.backupRowproducer = getKafkaProducer(kafkaServers);
+	}
+
+	private KafkaProducer<String, String> getKafkaProducer(String kafkaServers) {
+		Properties kafkaProperties = new Properties();
+		kafkaProperties.setProperty("bootstrap.servers", kafkaServers);
+		kafkaProperties.setProperty("key.serializer",  "org.apache.kafka.common.serialization.StringSerializer");
+		kafkaProperties.setProperty("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+		kafkaProperties.setProperty("acks", "1");
+		kafkaProperties.setProperty("retries", "5");
+		kafkaProperties.setProperty("batch.size", "20");
+
+		return new KafkaProducer<String, String>(kafkaProperties);
+	}
+
+	protected void flush()
+	{
+		if (!columnFamily.isEmpty())
+		{
+			//TODO, instead of printRow give a meaningfull name as it will put the purge data in kafka
+			nonLocalRowArchive(this::printCellConsumer);
+			columnFamily.clear();
+		}
+	}
+
+	private void printCellConsumer(Cell cell) {
+		if (cell != null)  {
+			Cell column = cell;
+			ColumnDefinition columnDefinition = columnFamily.metadata().getColumnDefinition(column.name());
+			if(columnDefinition == null) {
+				return;
+			}
+			if(columnDefinition.isPrimaryKeyColumn()) {
+				logger.info("primary key: {}", column.name().cql3ColumnName(columnFamily.metadata()).toString());
+				return;
+			}
+			try {
+				if (column.value() != null && column.value().array().length > 0) {
+					logger.debug("column identifier:{}", column.name().cql3ColumnName(columnFamily.metadata()).toString());
+				}
+			} catch (Exception e) {
+				logger.warn("Exception occurred while printing cell", e);
+			}
+		}
+	}
+
+	private void handleNonLocalArchiving(Cell cell, CassandraPurgedData purgedData) {
+		if (cell != null)  {
+			Cell column = cell;
+			ColumnDefinition columnDefinition = columnFamily.metadata().getColumnDefinition(column.name());
+			if(columnDefinition == null) {
+				return;
+			}
+			if(columnDefinition.isPrimaryKeyColumn()) {
+				purgedData.addPartitonKey(column.name().cql3ColumnName(columnFamily.metadata()).toString(),
+						SerializerMetaDataFactory.getSerializerMetaData(columnFamily.metadata().getColumnDefinition(column.name()).type.getSerializer()),
+						column.value(), Long.valueOf(column.timestamp()));
+				return;
+			} else {
+				purgedData.addNonKeyCell(SerializerMetaDataFactory.getSerializableCellData(cell, columnFamily), column.name().cql3ColumnName(columnFamily.metadata()).toString(), SerializerMetaDataFactory.getSerializerMetaData(columnFamily.metadata().getColumnDefinition(column.name()).type.getSerializer()));
+
+			}
+		}
+	}
+
+	private void nonLocalRowArchive(Consumer<Cell> printCell) {
```

Comment:
it is getting used when we call archiveRow, since it requires one consumer, so i created printCell, though we are not suing that consumer inside archiveRow,